core/src/main/java/io/aiven/kafka/tieredstorage/transform/CloseableEnumeration.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.transform;
+
+public abstract class CloseableEnumeration<T> implements java.util.Enumeration<T>, AutoCloseable {
```

Comment:
```suggestion
public abstract class CloseableEnumeration<T extends InputStream> implements java.util.Enumeration<T>, AutoCloseable {
```

Reply:
I don't think we should have tied to `InputStream` specifically since it can be reused for whatever else types in future if needed.